src/daft-functions/src/binary/concat.rs

```diff
@@ -0,0 +1,129 @@
+use common_error::{DaftError, DaftResult};
+use daft_core::{
+    array::ops::as_arrow::AsArrow,
+    datatypes::{BinaryArray, DataType, Field, FixedSizeBinaryArray},
+    prelude::Schema,
+    series::{IntoSeries, Series},
+};
+use daft_dsl::{
+    functions::{ScalarFunction, ScalarUDF},
+    ExprRef,
+};
+use serde::{Deserialize, Serialize};
+
+#[derive(Debug, Clone, Serialize, Deserialize, PartialEq, Eq, Hash)]
+pub struct BinaryConcat {}
+
+#[typetag::serde]
+impl ScalarUDF for BinaryConcat {
+    fn as_any(&self) -> &dyn std::any::Any {
+        self
+    }
+    fn name(&self) -> &'static str {
+        "concat"
+    }
+
+    fn to_field(&self, inputs: &[ExprRef], schema: &Schema) -> DaftResult<Field> {
+        match inputs {
+            [left, right] => {
+                let left_field = left.to_field(schema)?;
+                let right_field = right.to_field(schema)?;
+                match (&left_field.dtype, &right_field.dtype) {
+                    (DataType::Binary, DataType::Binary) => {
+                        Ok(Field::new(left_field.name, DataType::Binary))
+                    }
+                    (DataType::Binary, DataType::Null) | (DataType::Null, DataType::Binary) => {
+                        Ok(Field::new(left_field.name, DataType::Binary))
+                    }
+                    (DataType::FixedSizeBinary(size1), DataType::FixedSizeBinary(size2)) => Ok(
+                        Field::new(left_field.name, DataType::FixedSizeBinary(size1 + size2)),
+                    ),
+                    (DataType::FixedSizeBinary(_), DataType::Binary)
+                    | (DataType::Binary, DataType::FixedSizeBinary(_)) => {
+                        Ok(Field::new(left_field.name, DataType::Binary))
+                    }
+                    (DataType::FixedSizeBinary(_), DataType::Null)
+                    | (DataType::Null, DataType::FixedSizeBinary(_)) => {
+                        Ok(Field::new(left_field.name, DataType::Binary))
+                    }
+                    _ => Err(DaftError::TypeError(format!(
+                        "Expects inputs to concat to be binary, but received {} and {}",
+                        format_field_type_for_error(&left_field),
+                        format_field_type_for_error(&right_field),
+                    ))),
+                }
+            }
+            _ => Err(DaftError::SchemaMismatch(format!(
+                "Expected 2 input args, got {}",
+                inputs.len()
+            ))),
+        }
+    }
+
+    fn evaluate(&self, inputs: &[Series]) -> DaftResult<Series> {
```

Comment:
This function is doing similar logic to Series::binary_concat, why don't we just call that here?